reark/src/main/java/io/reark/reark/data/stores/cores/ContentProviderStoreCoreBase.java

```diff
@@ -59,25 +67,70 @@
  * @param <U> Type of the data this store core contains.
  */
 public abstract class ContentProviderStoreCoreBase<U> {
-    private static final String TAG = ContentProviderStoreCoreBase.class.getSimpleName();
+
+    private final String TAG = getClass().getSimpleName();
 
     @NonNull
-    private final ContentResolver contentResolver;
+    private static final ContentProviderOperation NO_OPERATION = ContentProviderOperation.newInsert(Uri.EMPTY).build();
 
     @NonNull
-    private final ContentObserver contentObserver = getContentObserver();
+    private final ContentResolver contentResolver;
 
     @NonNull
     private final PublishSubject<Pair<U, Uri>> updateSubject = PublishSubject.create();
 
+    @NonNull
+    private final ObjectLockHandler<Uri> locker = new ObjectLockHandler<>();
+
+    @Nullable
+    private Subscription updateSubscription;
+
+    private final int groupingTimeout;
+
+    private final int groupMaxSize;
+
     protected ContentProviderStoreCoreBase(@NonNull final ContentResolver contentResolver) {
+        this(contentResolver, 10, 30);
+    }
+
+    protected ContentProviderStoreCoreBase(@NonNull final ContentResolver contentResolver,
+                                           final int groupingTimeout,
+                                           final int groupMaxSize) {
         this.contentResolver = Preconditions.get(contentResolver);
-        this.contentResolver.registerContentObserver(getContentUri(), true, contentObserver);
+        this.groupingTimeout = groupingTimeout;
+        this.groupMaxSize = groupMaxSize;
 
-        updateSubject
-                .onBackpressureBuffer()
-                .observeOn(Schedulers.io())
-                .subscribe(this::updateIfValueChanged);
+        initialize();
+    }
+
+    private void initialize() {
+        this.contentResolver.registerContentObserver(getContentUri(), true, getContentObserver());
+
+        // Observable transforming updates and inserts to ContentProviderOperations
+        Observable<ContentProviderOperation> operationObservable = updateSubject
+                .flatMap(pair -> createOperation(pair.first, pair.second));
```

Comment:
The scheduler was chosen at the end of createOperation() with subscribeOn. Changed it to more explicit observeOn and moved here